Develop Reusable Components for Profile Page and Use API to Use Mock …

lhlee-umass opened the pull request:
- Add Profile Picture Component
- Add Username Handle Component
- Add Reward Points Component
- Add Star Rating Component
- Add Review Tile Component
- Add Verification Badges
- Use API to use mock data of Users and Reviews to test displaying profile page
- Temporarily route /profile to a profile page (clicking "Seller Profile" on messaging page will direct you to /profile)

This PR will result in finishing the following issues:
Closes https://github.com/lhlee-umass/streetside/issues/48 (main issue)
Closes https://github.com/lhlee-umass/streetside/issues/49 (sub-issue)
Closes https://github.com/lhlee-umass/streetside/issues/50 (sub-issue)
Closes https://github.com/lhlee-umass/streetside/issues/51 (sub-issue)
Closes https://github.com/lhlee-umass/streetside/issues/52 (sub-issue)
Closes https://github.com/lhlee-umass/streetside/issues/53 (sub-issue)
Closes https://github.com/lhlee-umass/streetside/issues/56 (sub-issue)

lhlee-umass committed:
Develop Reusable Components for Profile Page and Use API to Use Mock Data

- Add Profile Picture Component
- Add Username Handle Component
- Add Reward Points Component
- Add Star Rating Component
- Add Review Tile Component
- Add Verification Badges
- Use API to use mock data of Users and Reviews to test displaying profile page
- Temporarily route /profile to a profile page (clicking "Seller Profile" on messaging page will direct you to /profile)

zhandavidz commented on frontend/src/pages/styles.css:
Looks great! Try to see if you can replace the css with Tailwind to make the components more componentized and isolated

lhlee-umass replied:
Done!

zhandavidz commented on frontend/src/pages/ProfilePage.tsx:
```diff
@@ -0,0 +1,108 @@
+        {'☆'.repeat(emptyStars)}
+      </div>
+    )
+  }
```
Could you use your `<StarRating />` component here? I think it could make things more unified and componentized, especially if we decide to change how star ratings are rendered in the future.

lhlee-umass replied:
Done!

zhandavidz commented on frontend/src/components/profile/VerificationBadge.tsx:
```diff
@@ -0,0 +1,25 @@
+
+const VerificationBadge: React.FC<VerificationBadgeProps> = ({
+  isVerified,
+  badgeText,
```
The verification data from the API returns a list of verifications, so maybe it could be good to list those verifications? Like for example, a lot have a verification of `['UMass']` showing that they are verified UMass emails, so it could be cool to show that they are specifically verified as a part of `UMass`

lhlee-umass replied:
Done!

zhandavidz commented on frontend/src/components/profile/UsernameHandle.tsx:
```diff
@@ -0,0 +1,20 @@
+}) => {
+  return (
+    <div className="username-handle">
+      <span>{username}</span>
```
Just a thought: it could be fun to have a `@` in front of the username to make it clear that it is a username.

lhlee-umass replied:
Done!

zhandavidz commented on frontend/src/components/profile/StarRating.tsx:
```diff
@@ -0,0 +1,23 @@
+const StarRating: React.FC<StarRatingProps> = ({ rating, totalStars = 5 }) => {
+  const filledStars = Math.round(rating)
+  const stars = Array.from({ length: totalStars }, (_, index) =>
+    index < filledStars ? '★' : '☆'
```
Do you think it would be possible to display fractional stars? or maybe half stars at least?

lhlee-umass replied:
Done!

zhandavidz commented on frontend/src/components/profile/RewardPoints.tsx:
```diff
@@ -0,0 +1,15 @@
+const RewardPoints: React.FC<RewardPointsProps> = ({ points }) => {
+  return (
+    <div className="reward-points">
+      <h3>Reward Points: {points}</h3>
```
Do you think you could include maybe a gold coin as the "unit" for reward points? I think we may be using this component in the top bar to show reward points there as well, so it may be nice to have a little emoji or icon instead of the text "Reward Points", and then the profile page could separately add `"Reward Points: "` if it wants that text too :)

lhlee-umass replied:
Done!

zhandavidz commented on frontend/src/components/profile/ReviewTile.tsx:
```diff
@@ -0,0 +1,58 @@
+  // Render stars based on rating
+  const stars = Array.from({ length: 5 }, (_, index) => {
+    return index < rating ? '★' : '☆' // Filled star if index < rating, else empty star
+  }).join('') // Join stars to create a string
```
Could you use your `<StarRating />` component here? I think it could make things more unified and componentized, especially if we decide to change how star ratings are rendered in the future.

lhlee-umass replied:
Done!

zhandavidz commented on frontend/src/components/profile/ReviewTile.tsx:
```diff
@@ -0,0 +1,58 @@
+        <h3>Review ID: {reviewId}</h3>
+        <p>Reviewer ID: {reviewerId}</p>
+        <p>Reviewee ID: {revieweeId}</p>
+      </div>
```
We probably shouldn't display the IDs directly, instead we could use those IDs along with the Users API to fetch the name of the Reviewer

lhlee-umass replied:
Done!

zhandavidz commented on frontend/src/components/profile/ReviewTile.tsx:
```diff
@@ -0,0 +1,58 @@
+  rating: number
+  message: string
+  createdAt: string
+}
```
These ReviewTileProps look a lot like the plain `Review` type from the API, this component could just accept an entire Review as a prop and then be able to more easily adapt to changes to the Review datatype.

lhlee-umass replied:
Done!

zhandavidz commented on frontend/src/components/profile/ReviewTile.tsx:
```diff
@@ -0,0 +1,58 @@
+        </p>{' '}
+        {/* Display the stars here */}
+        <p>
+          <strong>Is Buyer:</strong> {reviewerIsBuyer ? 'Yes' : 'No'}
```
Just a thought: is there a cool way to integrate that into the display of the name of the reviewer? The idea behind this attribute of the review is to allow people to be reviewed both as a seller and as a buyer, so maybe if `reviewerIsBuyer` is true, we could display `"Review by buyer: [first name, last name]"` and if it is false, we could display `"Review by seller: [first name, last name]"`

lhlee-umass replied:
Done!

zhandavidz commented on frontend/src/components/profile/ProfilePicture.tsx:
```diff
@@ -0,0 +1,28 @@
+
+interface ProfilePictureProps {
+  imageUrl: string
+  altText: string
```
Could replace the `altText` prop with some a `username` or `name` prop that will then set the alt text of the image to "Profile Picture of: {altText}"

lhlee-umass replied:
Done!

zhandavidz reviewed (changes requested):
Great work! I left a few comments on some things that could be changed, let me know if you have any questions :)

lhlee-umass committed:
Requested Formatting Changes

- Replace altText prop with name in ProfilePicture.tsx
- Take in entire Review type as prop, use StarRating component, and remove reviewer/reviewee ID in ReviewTile.tsx
- Display gold coin in RewardPoints.tsx
- Ability to display half stars in reviews in StarRating.tsx
- Add @ in front of username in UsernameHandle.tsx
- Show what makes a user verified in VerificationBadge.tsx
- Use StarRating component for average rating in ProfilePage.tsx
- Change to Tailwind CSS and delete styles.css

zhandavidz reviewed (approved):
All these new changes look good! I think they fix all the problems noted earlier, and I didn't find any issues in testing either, LGTM!

I think it's ready to merge!
